packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -499,7 +499,8 @@ class DevFS {
         generator.reset();
       }
       final String compiledBinary =
-          await generator.recompile(mainPath, invalidatedFiles);
+          await generator.recompile(mainPath, invalidatedFiles,
+              outputPath: fs.path.join(getBuildDirectory(), 'app.dill'));
```

Comment:
Doesn't this depend on the value passed in to `--output-dill`?   e.g. in Google, we're going to build the .dill file to a different location.

Reply:
/cc @keertip